src/transformers/modeling_jax_utils.py

```diff
@@ -0,0 +1,145 @@
+import os
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from flax.nn import Model, Module
+from flax.serialization import to_bytes
+from flax.traverse_util import unflatten_dict
+from jax.random import PRNGKey
+
+from transformers import PretrainedConfig, logger
+from transformers.file_utils import hf_bucket_url, cached_path, WEIGHTS_NAME, TF2_WEIGHTS_NAME, TF_WEIGHTS_NAME, \
+    is_remote_url
+
+
+class JaxPreTrainedModel(ABC):
+    config_class = None
+    pretrained_model_archive_map = {}
+    base_model_prefix = ""
+    model_class = None
```

Comment:
What is `model_class` used for?

Reply:
`model_class` is the underlying flax.nn.Module class which is required by Flax/msgpack to allocated all the buffers.